src/commands/wp.ts

```diff
@@ -0,0 +1,190 @@
+/**
+ * External dependencies
+ */
+import debugLib from 'debug';
+import gql from 'graphql-tag';
+import * as ssh2 from 'ssh2';
+
+/**
+ * Internal dependencies
+ */
+import pkg from '../../package.json';
+import { App, AppEnvironment } from '../graphqlTypes';
+import API from '../lib/api';
+import { CommandTracker, makeCommandTracker } from '../lib/tracker';
+
+const debug = debugLib( '@automattic/vip:wp/ssh' );
+
+interface TriggerWPCLICommandMutationResponse {
+	triggerWPCLICommandOnAppEnvironment: {
+		inputToken: string;
+		sshAuthentication: {
+			host: string;
+			port: number;
+			username: string;
+			privateKey: string;
+			passphrase: string;
+		};
+		command: {
+			guid: string;
+		};
+	};
+}
+
+const TRIGGER_WP_CLI_COMMAND_MUTATION = gql`
+	mutation TriggerWPCLICommandMutation($input: AppEnvironmentTriggerWPCLICommandInput) {
+		triggerWPCLICommandOnAppEnvironment(input: $input) {
+			inputToken
+			sshAuthentication {
+				host
+				port
+				username
+				privateKey
+				passphrase
+			}
+			command {
+				guid
+			}
+		}
+	}
+`;
+
+const getSSHAuthForCommand = async ( appId: number, envId: number, command: string ) => {
+	const api = API();
+
+	return api.mutate( {
+		mutation: TRIGGER_WP_CLI_COMMAND_MUTATION,
+		variables: {
+			input: {
+				id: appId,
+				environmentId: envId,
+				command,
+			},
+		},
+	} ) as Promise< { data: TriggerWPCLICommandMutationResponse } >;
+};
+
+export class WPCliCommandOverSSH {
+	private app: App;
+	private env: AppEnvironment;
+	private track: CommandTracker;
+
+	constructor( app: App, env: AppEnvironment ) {
+		this.app = app;
+		this.env = env;
+
+		this.track = makeCommandTracker( 'wp', {
+			app: this.app.id,
+			env: this.env.id,
+			executionType: 'ssh',
```

Comment:
I think this needs to be snake case:

```
  @automattic/vip:analytics:clients:tracks Error: Invalid prop name detected: executionType -- this event will be rejected during ETL +10s
```

```suggestion
			execution_type: 'ssh',
```